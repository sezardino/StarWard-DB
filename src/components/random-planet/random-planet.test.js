import React from 'react';
import {shallow} from 'enzyme';

import RandomPlanet from './random-planet';

describe('Component RandomPlanet', () => {
	const mockData = {
		name: 'Test name',
		population: 'Test population',
		rotationPeriod: 'Test period',
		diameter: 'Test diameter',
		img: 'Test img',
	};
	const component = shallow(<RandomPlanet data={mockData} />);
	const componentWithoutData = shallow(<RandomPlanet />);

	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

	it('Component doest render without data', () => {
		expect(componentWithoutData.isEmptyRender()).toBe(true);
	});
});
