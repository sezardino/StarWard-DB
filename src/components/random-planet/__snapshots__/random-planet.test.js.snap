// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component RandomPlanet Component rendered correctly 1`] = `
"<div className=\\"random-planet jumbotron rounded\\">
  <img className=\\"planet-image\\" src=\\"Test img\\" alt=\\"Test name\\" />
  <div>
    <h4>
      Test name
    </h4>
    <ul className=\\"list-group list-group-flush\\">
      <li className=\\"list-group-item\\">
        <span className=\\"term\\">
          Population
        </span>
        <span>
          Test population
        </span>
      </li>
      <li className=\\"list-group-item\\">
        <span className=\\"term\\">
          Rotation Period
        </span>
        <span>
          Test period
        </span>
      </li>
      <li className=\\"list-group-item\\">
        <span className=\\"term\\">
          Diameter
        </span>
        <span>
          Test diameter
        </span>
      </li>
    </ul>
  </div>
</div>"
`;
